Python/thread_pthread.h

```diff
@@ -51,6 +51,12 @@
 #undef  THREAD_STACK_SIZE
 #define THREAD_STACK_SIZE       0x200000
 #endif
+#if defined(__ANDROID__) && defined(THREAD_STACK_SIZE) && THREAD_STACK_SIZE == 0
+#   ifdef Py_DEBUG
```

Comment:
Please add a comment explaining why this is needed. I understand the Python compiled in debug mode uses more stack and that test_threading.test_recursion_limit() requires to be able to call up to 1000 Python frames in a thread without crashing. I propose the comment:

```
/* bpo-38852: test_threading.test_recursion_limit() checks that 1000 recursive Python calls
   (default recursion limit) doesn't crash, but raise a regular RecursionError exception.
   In debug mode, Python function calls allocates more memory on the stack, so use a
   stack of 8 MiB. */
```